src/test/check_getdns_context_set_dns_transport.h

```diff
@@ -71,108 +71,222 @@
 
+       /* Re-do over TCP */
+       ASSERT_RC(getdns_context_set_dns_transport(context, GETDNS_TRANSPORT_TCP_ONLY),
+         GETDNS_RETURN_GOOD, "Return code from getdns_context_set_dns_transport()");   
 
-      ASSERT_RC(getdns_dict_get_int(ex_response.header, "ancount", &ancount),
-        GETDNS_RETURN_GOOD, "Failed to extract \"nscount\"");
+       ASSERT_RC(getdns_general_sync(context, "getdnsapi.net", 48, extensions, &response), 
+         GETDNS_RETURN_GOOD, "Return code from getdns_general_sync()");
 
-      ASSERT_RC(getdns_dict_get_int(ex_response.header, "arcount", &arcount),
-        GETDNS_RETURN_GOOD, "Failed to extract \"nscount\"");
+       ASSERT_RC(getdns_dict_get_int(response, "/call_reporting/0/transport", &transport),
+         GETDNS_RETURN_GOOD, "Failed to extract \"transport\"");
+       ASSERT_RC(transport, GETDNS_TRANSPORT_TCP, "Query did not go over TCP");
+       ASSERT_RC(getdns_dict_get_int(response, "/replies_tree/0/header/tc", &tc),
+         GETDNS_RETURN_GOOD, "Failed to extract \"tc\"");
+       ASSERT_RC(tc, 0, "Packet trucated - not as expected");
 
-      ASSERT_RC(getdns_dict_get_int(ex_response.header, "nscount", &nscount),
-        GETDNS_RETURN_GOOD, "Failed to extract \"nscount\"");
+       /* Now let it fall back to TCP */
+       ASSERT_RC(getdns_context_set_dns_transport(context, GETDNS_TRANSPORT_UDP_FIRST_AND_FALL_BACK_TO_TCP),
+         GETDNS_RETURN_GOOD, "Return code from getdns_context_set_dns_transport()");   
+       ASSERT_RC(getdns_general_sync(context, "getdnsapi.net", 48, extensions, &response), 
+         GETDNS_RETURN_GOOD, "Return code from getdns_general_sync()");
 
-       printf("the resp is %s\n", getdns_pretty_print_dict(response));
-       printf("the ancount is %d\n", ancount);
-       printf("the arcount is %d\n", arcount);
-       printf("the nscount is %d\n", nscount);
-       udp_sum = ancount + arcount + nscount;
-       printf("the udp_sum is %d\n", udp_sum);
+       ASSERT_RC(getdns_dict_get_int(response, "/call_reporting/0/transport", &transport),
+         GETDNS_RETURN_GOOD, "Failed to extract \"transport\"");
+       ASSERT_RC(transport, GETDNS_TRANSPORT_TCP, "Query did not go over TCP");
+       ASSERT_RC(getdns_dict_get_int(response, "/replies_tree/0/header/tc", &tc),
+         GETDNS_RETURN_GOOD, "Failed to extract \"tc\"");
+       ASSERT_RC(tc, 0, "Packet trucated - not as expected");
 
-       //tcp count
-       ASSERT_RC(getdns_context_set_dns_transport(context, GETDNS_TRANSPORT_TCP_ONLY),
-         GETDNS_RETURN_GOOD, "Return code from getdns_context_set_dns_transport()");   
-     
+      CONTEXT_DESTROY;
 
-       ASSERT_RC(getdns_general_sync(context, "google.com", 255, NULL, &response), 
-         GETDNS_RETURN_GOOD, "Return code from getdns_general_sync()");
+     }
+     END_TEST
 
-       struct extracted_response ex_response1;	
-       extract_response(response, &ex_response1);
+     START_TEST (getdns_context_set_dns_transport_recursing_6)
+     {
```

Comment:
This test fails in the 230-stub-only-run-unit-tests tpkg, because the default resolution type will stub mode and not recursion.  You could test for it by trying to set the resolution type to RECURSION explicitly and skip the whole thing is it returns GETDNS_RETURN_NOT_IMPLEMENTED.